### PR TITLE
Adds journal_lines stream

### PR DESCRIPTION
**Pull request by vmesel**
# Description of change
(write a short description or paste a link to JIRA)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_xero/streams.py [hunk 1]
```diff
@@ -191,6 +191,36 @@ def sync(self, ctx):
...
+                lines_stream_id,
+                lines_schema,
+                ["JournalLineID"]
+            )
+
+        for rec in records:
+            with Transformer() as transformer:
+                rec = transformer.transform(rec, schema, metadata.to_map(mdata))
+                singer.write_record(self.tap_stream_id, rec)
+            if "JournalLines" in rec and len(line_mdata) > 0:
```
Comment (hsyyid) [inline tap_xero/streams.py]: This `len(line_mdata)` will fail if `line_mdata` is `None`

File: tap_xero/streams.py [hunk 2]
```diff
@@ -191,6 +191,36 @@ def sync(self, ctx):
...
 
+    def write_records(self, records, ctx):
+        """"Custom implementation from the write records method available in Stream class"""
+        stream = ctx.catalog.get_stream(self.tap_stream_id)
+        schema = stream.schema.to_dict()
+        lines_schema = schema["properties"].get("JournalLines", {}).get("items")
+        lines_stream_id = "{}_lines".format(self.tap_stream_id)
+        mdata = stream.metadata
+        try:
+            line_mdata = [i for i in mdata if "JournalLines" in i.get("breadcrumb", [])]
```
Comment (hsyyid) [inline tap_xero/streams.py:202]: So we're just checking if `JournalLines` exists as a field in the metadata? Or are we actually adding a new stream?

**Review by hsyyid — CHANGES_REQUESTED** [review]

File: tap_xero/streams.py (hunk 1 above)
Comment (vmesel) [reply tap_xero/streams.py]: Solving.

File: tap_xero/streams.py (hunk 2 above)
Comment (vmesel) [reply tap_xero/streams.py:202]: We are checking if JournalLines exist in the record and in the metadata, then we are "adding" a new stream by sending the schema and record lines to the file, not creating a new stream class in itself

**Review by hsyyid — APPROVED** [review]

**Pull request merged**